odoo/addons/base/models/res_users.py

```diff
@@ -447,8 +447,9 @@ def onchange_parent_id(self):
 
     def _read(self, fields):
         super(Users, self)._read(fields)
-        canwrite = self.check_access_rights('write', raise_exception=False)
-        if not canwrite and set(USER_PRIVATE_FIELDS).intersection(fields):
+        if set(USER_PRIVATE_FIELDS).intersection(fields):
```

Comment:
Maybe
```python
if set(USER_PRIVATE_FIELDS).isdisjoint(fields):
    return
if self.check_access_rights('write', raise_exception=False):
    return
# cleanup here
```
would make for a clearer condition? And it would use `isdisjoint` which is cool.

The two guards could be combined with `or` as well, but I don't know if it's clearer this or that way.

Reply:
The idea was to have a minimal blame

Reply:
```python
if set(USER_PRIVATE_FIELDS).isdisjoint(fields) or self.check_access_rights('write', raise_exception=False):
    return
```
is a shorter blame than the current proposal, and the other alternative adds one line to the existing addition.